sklearn/utils/tests/test_validation.py

```diff
@@ -245,6 +245,22 @@ def test_check_array():
     result = check_array(X_no_array)
     assert_true(isinstance(result, np.ndarray))
 
+    # deprecation warning if string-like array with dtype="numeric"
+    X_str = [['a', 'b'], ['c', 'd']]
+    assert_warns(DeprecationWarning, check_array, X_str, "numeric")
```

Comment:
Please check the warning message through assert_warns_message.

Reply:
Thanks @lesteve, fixed in commit 50cd194